src/MoBi.Presentation/UICommand/AddNewExpressionProfileBuildingBlock.cs

```diff
@@ -0,0 +1,73 @@
+﻿using MoBi.Core.Domain.Model;
+using MoBi.Core.Services;
+using MoBi.Presentation.Tasks.Interaction;
+using OSPSuite.Core.Domain.Builder;
+using OSPSuite.Presentation.UICommands;
+
+namespace MoBi.Presentation.UICommand
+{
+   public abstract class AddExpressionProfileBuildingBlock : ObjectUICommand<AddExpressionProfileBuildingBlock>
+   {
+      protected IPKSimStarter _pkSimStarter;
+      private readonly IMoBiContext _context;
+      private readonly IInteractionTasksForExpressionProfileBuildingBlock _interactionTasksForExpressionProfileBuildingBlock;
+
+      protected AddExpressionProfileBuildingBlock(IPKSimStarter pkSimStarter, IMoBiContext context,
+         IInteractionTasksForExpressionProfileBuildingBlock interactionTasksForExpressionProfileBuildingBlock)
+      {
+         _pkSimStarter = pkSimStarter;
+         _context = context;
+         _interactionTasksForExpressionProfileBuildingBlock = interactionTasksForExpressionProfileBuildingBlock;
+      }
+
+      protected override void PerformExecute()
+      {
+         var expressionFromPKSim = CreateExpressionFromPKSim();
+         if (expressionFromPKSim != null)
```

Comment:
don't hesitate to exit early
https://github.com/Open-Systems-Pharmacology/Suite/blob/develop/CODING_STANDARDS.md#coding-style